aspnetcore/signalr/hubcontext.md

```diff
@@ -101,6 +102,40 @@ public class ChatController : Controller
 
 See [Strongly typed hubs](xref:signalr/hubs#strongly-typed-hubs) for more information.
 
+::: moniker range=">= aspnetcore-6.0"
+
+### Use IHubContext in generic code
+
+An injected `IHubContext<THub>` instance can be cast to a base hub type or the `Hub` type.
+`IHubContext<THub>` can also be cast to `IHubContext` without a generic `Hub` type specified.
+
+```csharp
+class BaseHub : Hub
+{ }
+
+class ChatHub : BaseHub
+{ }
+
+app.Use(async (context, next) =>
+{
+    var hubContext = context.RequestServices
+                            .GetRequiredService<IHubContext<ChatHub>>();
+    MethodA((IHubContext<BaseHub>)hubContext);
+
+    MethodB((IHubContext<Hub>)hubContext);
+
+    MethodC((IHubContext)hubContext);
```

Comment:
Why even bother demonstrating that the `(IHubContext<BaseHub>)` and `(IHubContext<Hub>)` casts are possible? It just adds unnecessary length to the conceptual docs when you can do all the same things by casting to `(IHubContext)` instead.

If someone really cares that it's covariant, they'll be able to see the `out THub` in the API docs, but it's suuuper niche.